charts/snowplow-iglu-server/templates/iglu-deployment.yaml

```diff
@@ -41,11 +41,8 @@ spec:
         imagePullPolicy: {{ default "IfNotPresent" .Values.service.image.pullPolicy }}
 
         args:
-        - "-Dconfig.override_with_env_vars=true"
-        {{- if ne .Values.service.config.hoconBase64 "" }}
         - "--config"
-        - "/etc/config/config.hocon"
-        {{- end }}
+        - {{if ne .Values.service.config.hoconBase64 "" }} "/etc/config/config.hocon" {{ else }} "/dev/null" {{ end }}
```

Comment:
Currently, yes it always needs to be specified.  We could change that, of course, but I'll explain why it's like this....

For all of our Snowplow/Iglu apps, we expect that the user provides a config file.  Technically, for all apps (collector, enrich, loaders, everything) it is possible to provide an empty config file and instead use environment variables or system properties.  But we don't realistically expect anyone to do that.  We only document the config file method.

So this helm stack is doing something we never expected to support, hence why it looks hacky.

I personally think this helm stack should use a kubernetes ConfigMap instead of using the `CONFIG_FORCE_*` trick.  In ECS we were forced to use the environment variables trick, because ECS has no nice way of mounting a config file.  But Kubernetes has great support for it, via ConfigMaps. 

I know you have a preference for doing all configuration by environment variables, if possible.  But I think we should be realistic that the config library we picked has much better support for config files rather than environment variables.

Reply:
@istreeter no issue in moving it to a configmap!  When we were using ECS configmaps became a bit of a pain but with kubernetes it is for sure a neater way to go!  If you want to have a crack at moving us towards that direction I am all for it.

Reply:
I do want to do it.  But I don't want to do it now.  My main objective now is to get something released that is compatible with distroless images, so I don't want to take on something extra that will slow me down.